File: Batch12/gaurav/array/MergeSortedArray.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int[] numsm = new int[m];
        int im=0;
        int jn=0;

        for(int i=0; i<m; i++) {
            numsm[i] = nums1[i];
        }

        int index=0;
        while(im < m && jn < n) {
            if(numsm[im] <= nums2[jn]) {
                nums1[index] = numsm[im];
                im++;
            }else{
                nums1[index] = nums2[jn];
                jn++;
            }
            index++;
        }

        for(int k=im; k<m; k++) {
            nums1[index] = numsm[k]; 
            index++;
        }

        for(int k=jn; k<n; k++) {
            nums1[index] = nums2[k]; 
            index++;
        }
    }
}

File: B13/Subhendu/Array/merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        ## https://leetcode.com/problems/merge-sorted-array/
        x = m - 1
        y = n - 1
        z = m + n - 1

        while x >= 0 and y >= 0:
            if nums1[x] > nums2[y]:
                nums1[z] = nums1[x]
                z -= 1
                x -= 1
            else:
                nums1[z] = nums2[y]
                z -= 1
                y -= 1

        while x >= 0:
            nums1[z] = nums1[x]
            z -= 1
            x -= 1

        while y >= 0:
            nums1[z] = nums2[y]
            z -= 1
            y -= 1

        return nums1
